src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -587,17 +642,90 @@ def docKeyUpdate(self, msOutDoc, **kwargs):
         for key, value in kwargs.items():
             try:
                 msOutDoc.setKey(key, value)
+                msOutDoc.updateTime()
             except Exception as ex:
                 msg = "Cannot update key {} for doc: {}\n".format(key, msOutDoc['_id'])
                 msg += "Error: {}".format(str(ex))
                 self.logger.warning(msg)
         return msOutDoc
 
-    def docInfoUpdate(self, msOutDoc):
+    def docInfoUpdate(self, msOutDoc, pipeLine=None):
         """
         A function intended to fetch and fill into the document all the needed
         additional information like campaignOutputMap etc.
         """
+
+        # Fill the destinationOutputMap first
+        if msOutDoc['isRelVal']:
+            destinationOutputMap = []
```

Comment:
and this logic should go under the template IMO. We provide a reqmgr2 document, we get the final mongodb document.

Reply:
Actually my understanding is exactly the opposite. My  origina idea was to have as minimum additional methods as possible in the Template and the ones that are there to be related only to validation checks and general components updates of the current object - in order to minimize the Memory footprint again. Because every additional method is about to be carried with the object through out it's whole lifetime. and once we start working in batches of Workflows at a time.... it may become noticeable.